src/test/java/org/opensearch/ad/bwc/ADBackwardsCompatibilityIT.java

```diff
@@ -83,25 +131,176 @@ public void testBackwardsCompatibility() throws Exception {
+                        categoryFieldSize
+                    );
+                    assertEquals(totalDocsPerCategory * categoryFieldSize * 2, getDocCountOfIndex(client(), dataIndexName));
                     Assert.assertTrue(pluginNames.contains("opendistro-anomaly-detection"));
                     Assert.assertTrue(pluginNames.contains("opendistro-job-scheduler"));
-                    createBasicAnomalyDetector();
+
+                    // Create single entity detector and start realtime job
+                    createRealtimeAnomalyDetectorsAndStart(SINGLE_ENTITY_DETECTOR);
+
+                    // Create single category HC detector and start realtime job
+                    createRealtimeAnomalyDetectorsAndStart(SINGLE_CATEGORY_HC_DETECTOR);
+
+                    // Create single entity historical detector and start historical analysis
+                    createHistoricalAnomalyDetectorsAndStart();
+
+                    // Verify cluster has 3 detectors now
+                    verifyAnomalyDetectorCount(TestHelpers.LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI, 3);
+                    // Verify cluster has 3 detectors with search detector API
+                    assertEquals(3, countDetectors(client(), null));
+
+                    // Verify all realtime detectors are running realtime job
+                    verifyRealtimeJobRunning();
                     break;
                 case MIXED:
+                    // We have no way to specify whether send request to old node or new node now.
+                    // TODO: Add more test later when it's possible to specify request node.
                     Assert.assertTrue(pluginNames.contains("opensearch-anomaly-detection"));
                     Assert.assertTrue(pluginNames.contains("opensearch-job-scheduler"));
-                    verifyAnomalyDetector(TestHelpers.LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI);
+
+                    // Verify cluster has 3 detectors now
+                    verifyAnomalyDetectorCount(TestHelpers.LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI, 3);
+                    // Verify cluster has 3 detectors with search detector API
+                    assertEquals(3, countDetectors(client(), null));
+
+                    // Verify all realtime detectors are running realtime job
+                    verifyRealtimeJobRunning();
                     break;
                 case UPGRADED:
                     Assert.assertTrue(pluginNames.contains("opensearch-anomaly-detection"));
                     Assert.assertTrue(pluginNames.contains("opensearch-job-scheduler"));
-                    verifyAnomalyDetector(TestHelpers.AD_BASE_DETECTORS_URI);
+
+                    Map<ADRestTestUtils.DetectorType, String> detectors = new HashMap<>();
+
+                    // Create single entity detector and start realtime job
+                    List<String> singleEntityDetectorResults = createRealtimeAnomalyDetectorsAndStart(SINGLE_ENTITY_DETECTOR);
```

Comment:
The `createRealtimeAnomalyDetectorsAndStart` ultimately calls the `startAnomalyDetector` which uses the `LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI` to start detector. If the cluster is `UPGRADED`, should we not use the `AD_BASE_DETECTORS_URI` instead?

Reply:
And if this works for upgraded cluster, wouldn't this work for mixed cluster as well where you can send request to `LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI` for creating new detectors?

Reply:
> The createRealtimeAnomalyDetectorsAndStart ultimately calls the startAnomalyDetector which uses the LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI to start detector. If the cluster is UPGRADED, should we not use the AD_BASE_DETECTORS_URI instead?

Testing for new URL `AD_BASE_DETECTORS_URI` has been covered in other REST test class like `AnomalyDetectorRestApiIT`. Here BWC is just for testing if old API still compatible on new cluster. We can add more testing to cover new URIs, but don't think that's our first priority in current phase as it has already been covered. 

> And if this works for upgraded cluster, wouldn't this work for mixed cluster as well where you can send request to LEGACY_OPENDISTRO_AD_BASE_DETECTORS_URI for creating new detectors?

For AD old version(<=1.0) we only support single flow and single-category HC detectors. For AD new version (>=1.1), we support new detector type: multi-category HC detector. So for multi-catetory HC detector, we should dispatch to new node, otherwise it will fail on old node. As we have no way to dispatch request to specific node, I would push this case to next release.

Ideally we should be able to create single flow and single-category HC detectors on mixed cluster no matter the request served by old or new node. But the mixed cluster has some tricky part for random detector name. I'm working on fixing that. But today is our cut off day for 1.1. So I wish to push what we have to 1.1 first so we can cover some BWC happy case at least automatically. Agree that we should cover more cases, but we should also develop agilely rather than waiting for all cases covered.

Reply:
That sounds good, thanks for the explanation @ylwu-amzn!

Reply:
@VachaShah thanks for your help. I added starting realtime job for single-flow detector and single-category HC detector on mixed cluster by checking which round current test is running.

Reply:
No problem! Thanks for adding.